### PR TITLE
5637 time based log rollover

### PR DESCRIPTION
**Pull request by ellen-lau**
Issue: https://github.com/OpenLiberty/docs/issues/5637

### CODE REVIEW

**Comment by ellen-lau** [comment]
@OpenLiberty/docs-members I see that @dmuelle is on vacation this week, could I get a review/approval from another reviewer? Thanks!

File: modules/ROOT/pages/access-logging.adoc [hunk 1]
```diff
@@ -142,3 +142,38 @@ Based on this setting, the NCSA access log includes the following information fo
...
+
+For example, a server with a rollover start time of midnight and a rollover interval of 1 day will roll over its HTTP access log once every day, at midnight.
+
+Enable time-based rollover of your HTTP access log file by using the `rolloverStartTime` and `rolloverInterval` attributes of the `httpAcccessLogging` or `accessLogging` elements in the `server.xml` file. The following table lists the two attributes, and their respective descriptions and permitted values.
+
+|===
+|Time-based log rollover attribute|Description
+
+|rolloverStartTime
+|Use this setting alone or with `rolloverInterval` to enable time-based log rollover for your HTTP access log file. This setting specifies the scheduled time of day for logs to first roll over. The `rolloverInterval` setting duration begins at `rolloverStartTime`. Valid values follow a 24-hour ISO-8601 datetime format of HH:MM, where 00:00 represents midnight. Padding zeros are required. If `rolloverInterval` is specified, the default value of `rolloverStartTime` is 00:00, midnight.
```
Comment (ramkumar-k-9286) [inline modules/ROOT/pages/access-logging.adoc]:
```suggestion
|Use this setting alone or with `rolloverInterval` attribute to enable time-based log rollover for your HTTP access log file. This setting specifies the scheduled time of day for logs to first rollover. The `rolloverInterval` setting duration begins at `rolloverStartTime`. Valid values follow a 24-hour ISO-8601 date-time format of HH:MM, where 00:00 represents midnight. Padding zeros are required. If `rolloverInterval` attribute is specified, the default value of `rolloverStartTime` attribute is 00:00, midnight.
```

File: modules/ROOT/pages/access-logging.adoc [hunk 2]
```diff
@@ -142,3 +142,38 @@ Based on this setting, the NCSA access log includes the following information fo
...
+
+|rolloverInterval
+| Use this setting alone or with `rolloverStartTime` to enable time-based log rollover for your HTTP access log file. This setting specifies the time interval in between log rollovers, in minutes if a unit of time is not specified. Specify a positive integer followed by a unit of time, which can be days (d), hours (h), or minutes (m). For example, specify 5 hours as 5h. You can include multiple values in a single entry. For example, 1d5h is equivalent to 1 day and 5 hours. If `rolloverStartTime` is specified, the default value of `rolloverInterval` is 1 day.
+
+|===
+
+The following examples have time-based periodic rollover enabled for the HTTP access log file in the `httpAcccessLogging` and `accessLogging` elements, with `rolloverStartTime` set to  midnight and `rolloverInterval` set to 1 day:
```
Comment (ramkumar-k-9286) [inline modules/ROOT/pages/access-logging.adoc]:
```suggestion
The following examples have time-based periodic rollover enabled for the HTTP access log file in the `httpAcccessLogging` and `accessLogging` elements, with `rolloverStartTime` attribute set to midnight and `rolloverInterval` attribute set to 1 day:
```

File: modules/ROOT/pages/access-logging.adoc [hunk 3]
```diff
@@ -142,3 +142,38 @@ Based on this setting, the NCSA access log includes the following information fo
...
+
+|===
+|Time-based log rollover attribute|Description
+
+|rolloverStartTime
+|Use this setting alone or with `rolloverInterval` to enable time-based log rollover for your HTTP access log file. This setting specifies the scheduled time of day for logs to first roll over. The `rolloverInterval` setting duration begins at `rolloverStartTime`. Valid values follow a 24-hour ISO-8601 datetime format of HH:MM, where 00:00 represents midnight. Padding zeros are required. If `rolloverInterval` is specified, the default value of `rolloverStartTime` is 00:00, midnight.
+
+|rolloverInterval
+| Use this setting alone or with `rolloverStartTime` to enable time-based log rollover for your HTTP access log file. This setting specifies the time interval in between log rollovers, in minutes if a unit of time is not specified. Specify a positive integer followed by a unit of time, which can be days (d), hours (h), or minutes (m). For example, specify 5 hours as 5h. You can include multiple values in a single entry. For example, 1d5h is equivalent to 1 day and 5 hours. If `rolloverStartTime` is specified, the default value of `rolloverInterval` is 1 day.
```
Comment (ramkumar-k-9286) [inline modules/ROOT/pages/access-logging.adoc]:
```suggestion
| Use this setting alone or with `rolloverStartTime` attribute to enable time-based log rollover for your HTTP access log file. This setting specifies the time interval in between log rollovers, in minutes if a unit of time is not specified. Specify a positive integer followed by a unit of time, which can be days (d), hours (h), or minutes (m). For example, specify 5 hours as 5h. You can include multiple values in a single entry. For example, 1d5h is equivalent to 1 day and 5 hours. If `rolloverStartTime` attribute is specified, the default value of `rolloverInterval` attribute is 1 day.
```

File: modules/ROOT/pages/log-trace-configuration.adoc [hunk 4]
```diff
@@ -300,6 +301,39 @@ You can use the `binaryLog` command to convert the binary log to a text file, as
 binaryLog view defaultServer
 ----
 
+[#rollover]
+== Time-based log rollover
+
+You can enable time-based periodic rollover of Liberty messages and trace logs by specificing a log rollover start time and a log rollover interval duration. The specified rollover start time is your desired time of day when logs should first roll over. The rollover interval duration is the time interval in between consecutive log rollovers.
```
Comment (ramkumar-k-9286) [inline modules/ROOT/pages/log-trace-configuration.adoc]:
```suggestion
You can enable time-based periodic rollover of Liberty messages and trace logs by specifying a log rollover start time and a log rollover interval duration. The specified rollover start time is your desired time of day when logs should first rollover. The rollover interval duration is the time interval between consecutive log rollovers.
```

File: modules/ROOT/pages/log-trace-configuration.adoc [hunk 5]
```diff
@@ -440,6 +474,16 @@ If you specify a value of `false`, the date and time are formatted according to
 |WLP_LOGGING_STACK_TRACE_SINGLE_ENTRY
 | When this attribute is set to `true`, stack traces that are emitted to standard-out and standard-error are merged into a single log event. By default, this value is set to `false` and each individual line of the stack trace is emitted as an individual log event.
 
+|rolloverStartTime
+|com.ibm.ws.logging.rollover.start.time
+|WLP_LOGGING_ROLLOVER_START_TIME
+| Use this setting alone or with `rolloverInterval` to enable time-based log rollover for `messages.log` and `trace.log` files. This setting specifies the scheduled time of day for logs to first roll over. The `rolloverInterval` setting duration begins at `rolloverStartTime`. Valid values follow a 24-hour ISO-8601 datetime format of HH:MM, where 00:00 represents midnight. Padding zeros are required. If `rolloverInterval` is specified, the default value of `rolloverStartTime` is 00:00, midnight. The `console.log` does not roll so this setting does not apply.
```
Comment (ramkumar-k-9286) [inline modules/ROOT/pages/log-trace-configuration.adoc]:
```suggestion
| Use this setting alone or with `rolloverInterval` attribute to enable time-based log rollover for `messages.log` and `trace.log` files. This setting specifies the scheduled time of day for logs to first rollover. The `rolloverInterval` setting duration begins at `rolloverStartTime`. Valid values follow a 24-hour ISO-8601 date-time format of HH:MM, where 00:00 represents midnight. Padding zeros are required. If `rolloverInterval` attribute is specified, the default value of `rolloverStartTime` attribute is 00:00, midnight. The `console.log` does not roll so this setting does not apply.
```

File: modules/ROOT/pages/log-trace-configuration.adoc [hunk 6]
```diff
@@ -440,6 +474,16 @@ If you specify a value of `false`, the date and time are formatted according to
...
+
+|rolloverInterval
+|com.ibm.ws.logging.rollover.interval
+|WLP_LOGGING_ROLLOVER_INTERVAL
+| Use this setting alone or with `rolloverStartTime` to enable time-based log rollover for `messages.log` and `trace.log` files. This setting specifies the time interval in between log rollovers, in minutes if a unit of time is not specified. Specify a positive integer followed by a unit of time, which can be days (d), hours (h), or minutes (m). For example, specify 5 hours as 5h. You can include multiple values in a single entry. For example, 1d5h is equivalent to 1 day and 5 hours. If `rolloverStartTime` is specified, the default value of `rolloverInterval` is 1 day. The `console.log` does not roll so this setting does not apply.
```
Comment (ramkumar-k-9286) [inline modules/ROOT/pages/log-trace-configuration.adoc]:
```suggestion
| Use this setting alone or with `rolloverStartTime` attribute to enable time-based log rollover for `messages.log` and `trace.log` files. This setting specifies the time interval in between log rollovers, in minutes if a unit of time is not specified. Specify a positive integer followed by a unit of time, which can be days (d), hours (h), or minutes (m). For example, specify 5 hours as 5h. You can include multiple values in a single entry. For example, 1d5h is equivalent to 1 day and 5 hours. If `rolloverStartTime` attribute is specified, the default value of `rolloverInterval` attribute is 1 day. The `console.log` does not roll so this setting does not apply.
```

**Review by ramkumar-k-9286 — APPROVED** [review]

**Comment by ellen-lau** [comment]
@OpenLiberty/docs-members can this be merged? I do not have merge authorization. Thanks!

File: modules/ROOT/pages/access-logging.adoc [hunk 7]
```diff
@@ -142,3 +142,38 @@ Based on this setting, the NCSA access log includes the following information fo
 * The response size in bytes excluding headers
 * The elapsed time of the request in microseconds, end-to-end, including client and network time
 * The elapsed time in microseconds until the first bytes of the response are sent. This value is often a close approximation of application response time.
+
+
+== Time-based HTTP access log rollover
+
+You can enable time-based periodic rollover of your HTTP access log file by specifying a log rollover start time and a log rollover interval duration. The specified rollover start time is your desired time of day when the HTTP access log file should first roll over. The rollover interval duration is the time interval in between consecutive HTTP access log file rollovers.
```
Comment (dmuelle) [inline modules/ROOT/pages/access-logging.adoc]:
```suggestion
You can enable time-based periodic rollover of your HTTP access log file by specifying a log rollover start time and a log rollover interval duration. The specified rollover start time is the time of day when the HTTP access log file first rolls over. The rollover interval duration is the time interval between consecutive HTTP access log file rollovers.
```

File: modules/ROOT/pages/access-logging.adoc [hunk 8]
```diff
@@ -142,3 +142,38 @@ Based on this setting, the NCSA access log includes the following information fo
...
+
+For example, a server with a rollover start time of midnight and a rollover interval of 1 day will roll over its HTTP access log once every day, at midnight.
```
Comment (dmuelle) [inline modules/ROOT/pages/access-logging.adoc]:
```suggestion
For example, a server with a rollover start time of midnight and a rollover interval of 1 day rolls over the HTTP access log once every day, at midnight.
```

File: modules/ROOT/pages/access-logging.adoc [hunk 9]
```diff
@@ -142,3 +142,38 @@ Based on this setting, the NCSA access log includes the following information fo
...
+
+Enable time-based rollover of your HTTP access log file by using the `rolloverStartTime` and `rolloverInterval` attributes of the `httpAcccessLogging` or `accessLogging` elements in the `server.xml` file. The following table lists the two attributes, and their respective descriptions and permitted values.
```
Comment (dmuelle) [inline modules/ROOT/pages/access-logging.adoc]:
```suggestion
Enable time-based rollover of your HTTP access log file by using the `rolloverStartTime` and `rolloverInterval` attributes of the `httpAcccessLogging` or `accessLogging` elements in the `server.xml` file. The following table lists the two attributes, their respective descriptions, and their permitted values.
```

File: modules/ROOT/pages/access-logging.adoc [hunk 10]
```diff
@@ -142,3 +142,38 @@ Based on this setting, the NCSA access log includes the following information fo
...
+
+|===
+|Time-based log rollover attribute|Description
+
+|rolloverStartTime
```
Comment (dmuelle) [inline modules/ROOT/pages/access-logging.adoc]:
```suggestion
|`rolloverStartTime`
```

File: modules/ROOT/pages/access-logging.adoc [hunk 11]
```diff
@@ -142,3 +142,38 @@ Based on this setting, the NCSA access log includes the following information fo
...
+|Use this setting alone or with the `rolloverInterval` attribute to enable time-based log rollover for your HTTP access log file. This setting specifies the scheduled time of day for logs to first rollover. The `rolloverInterval` setting duration begins at `rolloverStartTime`. Valid values follow a 24-hour ISO-8601 date-time format of HH:MM, where 00:00 represents midnight. Padding zeros are required. If the `rolloverInterval` attribute is specified, the default value of the `rolloverStartTime` attribute is 00:00, midnight.
+
+|rolloverInterval
```
Comment (dmuelle) [inline modules/ROOT/pages/access-logging.adoc]:
```suggestion
|`rolloverInterval`
```

File: modules/ROOT/pages/access-logging.adoc [hunk 12]
```diff
@@ -142,3 +142,38 @@ Based on this setting, the NCSA access log includes the following information fo
...
+| Use this setting alone or with the `rolloverStartTime` attribute to enable time-based log rollover for your HTTP access log file. This setting specifies the time interval in between log rollovers, in minutes if a unit of time is not specified. Specify a positive integer followed by a unit of time, which can be days (d), hours (h), or minutes (m). For example, specify 5 hours as 5h. You can include multiple values in a single entry. For example, 1d5h is equivalent to 1 day and 5 hours. If the `rolloverStartTime` attribute is specified, the default value of the `rolloverInterval` attribute is 1 day.
+
+|===
+
+The following examples have time-based periodic rollover enabled for the HTTP access log file in the `httpAcccessLogging` and `accessLogging` elements, with the `rolloverStartTime` attribute set to  midnight and the `rolloverInterval` attribute set to 1 day:
```
Comment (dmuelle) [inline modules/ROOT/pages/access-logging.adoc]:
```suggestion
The following examples demonstrate how to enable time-based periodic rollover for the HTTP access log file in the `httpAcccessLogging` and `accessLogging` elements, with the `rolloverStartTime` attribute set to  midnight and the `rolloverInterval` attribute set to 1 day.
```

File: modules/ROOT/pages/log-trace-configuration.adoc [hunk 13]
```diff
@@ -300,6 +301,39 @@ You can use the `binaryLog` command to convert the binary log to a text file, as
 binaryLog view defaultServer
 ----
 
+[#rollover]
+== Time-based log rollover
+
+You can enable time-based periodic rollover of Liberty messages and trace logs by specifying a log rollover start time and a log rollover interval duration. The specified rollover start time is your desired time of day when logs should first rollover. The rollover interval duration is the time interval in between consecutive log rollovers.
```
Comment (dmuelle) [inline modules/ROOT/pages/log-trace-configuration.adoc]:
```suggestion
You can enable time-based periodic rollover of Liberty messages and trace logs by specifying a log rollover start time and a log rollover interval duration. The specified rollover start time is the time of day when logs first roll over. The rollover interval duration is the time interval in between consecutive log rollovers.
```

File: modules/ROOT/pages/log-trace-configuration.adoc [hunk 14]
```diff
@@ -300,6 +301,39 @@ You can use the `binaryLog` command to convert the binary log to a text file, as
...
+
+For example, a server with a rollover start time of midnight and a rollover interval of 1 day will roll over its messages and trace logs once every day, at midnight.
```
Comment (dmuelle) [inline modules/ROOT/pages/log-trace-configuration.adoc]:
```suggestion
For example, a server with a rollover start time of midnight and a rollover interval of 1 day rolls over the messages and trace logs once every day, at midnight.
```

File: modules/ROOT/pages/log-trace-configuration.adoc [hunk 15]
```diff
@@ -300,6 +301,39 @@ You can use the `binaryLog` command to convert the binary log to a text file, as
...
+
+You can configure rollover for Open Liberty messages and trace logs by either editing the `bootstrap.properties` file, specifying environment variables, or using server configuration attributes. The following examples show the configuration for each of these options, with a log rollover start time of midnight and a log rollover interval of 1 day:
```
Comment (dmuelle) [inline modules/ROOT/pages/log-trace-configuration.adoc]:
```suggestion
You can configure rollover for Open Liberty messages and trace logs by either editing the `bootstrap.properties` file, specifying environment variables, or using server configuration attributes. The following examples show the configuration for each of these options, with a log rollover start time of midnight and a log rollover interval of 1 day.
```

File: modules/ROOT/pages/log-trace-configuration.adoc [hunk 16]
```diff
@@ -300,6 +301,39 @@ You can use the `binaryLog` command to convert the binary log to a text file, as
...
+
+* Configure time-based log rollover in the `bootstrap.properties` file:
```
Comment (dmuelle) [inline modules/ROOT/pages/log-trace-configuration.adoc]:
```suggestion
* The following example configures time-based log rollover in the `bootstrap.properties` file.
```

File: modules/ROOT/pages/log-trace-configuration.adoc [hunk 17]
```diff
@@ -300,6 +301,39 @@ You can use the `binaryLog` command to convert the binary log to a text file, as
...
++
+[source,properties,linenums,role="code_column"]
+----
+com.ibm.ws.logging.rollover.start.time=00:00
+com.ibm.ws.logging.rollover.interval=1m
+----
++
+* Configure time-based log rollover with environment variables in the `server.env` file:
```
Comment (dmuelle) [inline modules/ROOT/pages/log-trace-configuration.adoc]:
```suggestion
* The following example configures time-based log rollover with environment variables in the `server.env` file.
```

File: modules/ROOT/pages/log-trace-configuration.adoc [hunk 18]
```diff
@@ -300,6 +301,39 @@ You can use the `binaryLog` command to convert the binary log to a text file, as
...
++
+[source,properties,linenums,role="code_column"]
+----
+WLP_LOGGING_ROLLOVER_START_TIME=00:00
+WLP_LOGGING_ROLLOVER_INTERVAL=1m
+----
++
+* Configure time-based log rollover with server configuration attributes in the `server.xml` file:
```
Comment (dmuelle) [inline modules/ROOT/pages/log-trace-configuration.adoc]:
```suggestion
* The following example configures time-based log rollover with server configuration attributes in the `server.xml` file.
```

File: modules/ROOT/pages/log-trace-configuration.adoc [hunk 19]
```diff
@@ -440,6 +474,16 @@ If you specify a value of `false`, the date and time are formatted according to
 |WLP_LOGGING_STACK_TRACE_SINGLE_ENTRY
 | When this attribute is set to `true`, stack traces that are emitted to standard-out and standard-error are merged into a single log event. By default, this value is set to `false` and each individual line of the stack trace is emitted as an individual log event.
 
+|rolloverStartTime
+|com.ibm.ws.logging.rollover.start.time
+|WLP_LOGGING_ROLLOVER_START_TIME
+| Use this setting alone or with the `rolloverInterval` attribute to enable time-based log rollover for `messages.log` and `trace.log` files. This setting specifies the scheduled time of day for logs to first rollover. The `rolloverInterval` setting duration begins at `rolloverStartTime`. Valid values follow a 24-hour ISO-8601 date-time format of HH:MM, where 00:00 represents midnight. Padding zeros are required. If the `rolloverInterval` attribute is specified, the default value of the `rolloverStartTime` attribute is 00:00, midnight. The `console.log` does not roll so this setting does not apply.
```
Comment (dmuelle) [inline modules/ROOT/pages/log-trace-configuration.adoc]:
```suggestion
| Use this setting alone or with the `rolloverInterval` attribute to enable time-based log rollover for `messages.log` and `trace.log` files. This setting specifies the scheduled time of day for logs to first roll over. The `rolloverInterval` setting duration begins at the time that is specified by the  `rolloverStartTime` attribute. Valid values follow a 24-hour ISO-8601 date-time format of HH:MM, where 00:00 represents midnight. Padding zeros are required. If the `rolloverInterval` attribute is specified, the default value of the `rolloverStartTime` attribute is 00:00, midnight. The `console.log` does not roll so this setting does not apply.
```

File: modules/ROOT/pages/access-logging.adoc [hunk 20]
```diff
@@ -142,3 +142,38 @@ Based on this setting, the NCSA access log includes the following information fo
...
+
+For example, a server with a rollover start time of midnight and a rollover interval of 1 day will roll over its HTTP access log once every day, at midnight.
+
+Enable time-based rollover of your HTTP access log file by using the `rolloverStartTime` and `rolloverInterval` attributes of the `httpAcccessLogging` or `accessLogging` elements in the `server.xml` file. The following table lists the two attributes, and their respective descriptions and permitted values.
+
+|===
+|Time-based log rollover attribute|Description
+
+|rolloverStartTime
+|Use this setting alone or with the `rolloverInterval` attribute to enable time-based log rollover for your HTTP access log file. This setting specifies the scheduled time of day for logs to first rollover. The `rolloverInterval` setting duration begins at `rolloverStartTime`. Valid values follow a 24-hour ISO-8601 date-time format of HH:MM, where 00:00 represents midnight. Padding zeros are required. If the `rolloverInterval` attribute is specified, the default value of the `rolloverStartTime` attribute is 00:00, midnight.
```
Comment (dmuelle) [inline modules/ROOT/pages/access-logging.adoc]:
```suggestion
|Use this setting alone or with the `rolloverInterval` attribute to enable time-based log rollover for your HTTP access log file. This setting specifies the scheduled time of day for logs to first roll over. The `rolloverInterval` setting duration begins at `rolloverStartTime`. Valid values follow a 24-hour ISO-8601 date-time format of HH:MM, where 00:00 represents midnight. Padding zeros are required. If the `rolloverInterval` attribute is specified, the default value of the `rolloverStartTime` attribute is 00:00, midnight.
```

**Pull request merged**